core/commands/add.go

```diff
@@ -29,6 +31,7 @@ const progressReaderIncrement = 1024 * 256
 const (
 	progressOptionName = "progress"
 	wrapOptionName     = "wrap-with-directory"
+	hiddenOptionName   = "hidden"
```

Comment:
On Thu, May 07, 2015 at 05:56:13AM -0700, Christian Couder wrote:
> > @@ -29,6 +31,7 @@ const progressReaderIncrement = 1024 * 256
> >  const (
> >  	progressOptionName = "progress"
> >  	wrapOptionName     = "wrap-with-directory"
> > +	hiddenOptionName   = "hidden"
> 
> I am not very found of talking about "hidden" files. It think it is
> confusing to talk about both "ignored" and "hidden" files.

I agree.  Folks can always add .* to their ignore list if they don't
want to track dot-prefixed files.  I understand that this makes things
a bit wonky on Windows (where we'd be ignoring FILE_ATTRIBUTE_HIDDEN),
but I'm not sure that supporting that attribute is worth a parallel
parameter.  Maybe we could have a magic-comment in .ipfsignore to
ignore that attribute?

  $ cat .ipfsignore
  # windows: hidden

> If we want some files or directories to be ignored by default, then
> `ipfs init` should create a default `.ipfsignore` file that contains
> lines to ignore those files, though it is true that there may be a
> few hardcoded files or directories that are always ignored (for
> example in Git the `.git` directory is obviously always ignored).

I'm not in favor of auto-creating an .ipfsignore in `ipfs init`, since
it's just extra weight for folks who don't need to ignore anything.

I'm also not in favor of hard-coded default ignores, since it seems
more consistent to track everything by default.  I don't want to be in
the business of picking reasonable defaults for all users, and it's
easy enough for folks to add things they want ignored to their
per-user or per-tree ignore lists.

Since we already have a per-user ~/.ipfs directory, I think it's worth
calling the per-user file ~/.ipfs/ignore and only using .ipfsignore
for in-tree ignore configuration.

> By the way it is good anyway if `ipfs init` creates `.ipfsignore`
> even if it is empty, but it's even better if it is created with some
> comments that explain how it works.

As I said above, I'd rather not create the file automatically.  But
docs are great, so I'd like some explaining how the ignore feature
works, the ignore-file syntax, etc.  It looks like the tour has a
stubby FileBasics chapter.  Maybe a new entry there?  At least until
we get the Merkle archive and can shift the tour into a bundle
(#1195).